lib/galaxy/visualization/plugins/resource_parser.py

```diff
@@ -0,0 +1,228 @@
+"""
+Deserialize Galaxy resources (hdas, ldas, datasets, genomes, etc.) from
+a dictionary of string data/ids (often from a query string).
+"""
+import weakref
+
+import galaxy.exceptions
+import galaxy.util
+from galaxy.util import bunch
+from galaxy.managers import visualizations as visualization_manager
+from galaxy.managers import hdas as hda_manager
+
+import logging
+log = logging.getLogger( __name__ )
+
+
+class ResourceParser( object ):
+    """
+    Given a parameter dictionary (often a converted query string) and a
+    configuration dictionary (curr. only VisualizationsRegistry uses this),
+    convert the entries in the parameter dictionary into resources (Galaxy
+    models, primitive types, lists of either, etc.) and return
+    in a new dictionary.
+
+    The keys used to store the new values can optionally be re-mapped to
+    new keys (e.g. dataset_id="NNN" -> hda=<HistoryDatasetAsscoation>).
+    """
+    primitive_parsers = {
+        'str'   : lambda param: galaxy.util.sanitize_html.sanitize_html( param, 'utf-8' ),
+        'bool'  : lambda param: galaxy.util.string_as_bool( param ),
+        'int'   : lambda param: int( param ),
```

Comment:
I realize this isn't new code, but I don't understand the utility of the lambda here and I think we should only use lambdas (with explanation!) when it really makes code simpler.   Is there a reason this isn't just 'int' :  int ?